File: exercise/1528-Shuffle-String/Mister-Hope.ts
/*
 * Runtime: 92 ms, faster than 72.15% of TypeScript online submissions for Shuffle String.
 *
 * Memory Usage: 40.4 MB, less than 35.44% of TypeScript online submissions for Shuffle String.
 */
const restoreString = (s: string, indices: number[]): string => {
  const result: string[] = [];

  for (let i = 0; i < indices.length; i++) result[indices[i]] = s.charAt(i);

  return result.join("");
};

File: leetcode/1528-Shuffle-String/Mister-Hope.java
/*
 * Runtime: 1 ms, faster than 99.95% of Java online submissions for Shuffle String.
 *
 * Memory Usage: 39.4 MB, less than 7.86% of Java online submissions for Shuffle String.
 */

class Solution {
  public String restoreString(String s, int[] indices) {
    char[] chars = new char[s.length()];

    for (int i = 0; i < indices.length; i++) chars[indices[i]] = s.charAt(i);

    return new String(chars);
  }
}
